### PR TITLE
[example] PR apply for an example of differential algorithm

### PR DESCRIPTION
**Pull request by Nanase-Nishino**

### CODE REVIEW

**Comment by neozhaoliang** [comment]
Please add some meta and background information about your example:
1. what's the algorithm animating?
2. which feature you want to highlight with this example?
3. where to find the reference, if user want to learn more about this algorithm?

**Comment by Nanase-Nishino** [comment]
> 
1.what does this script animate?
This script reflects the optimization process of differential evolution. The black points are the search agents and they find out the minimum solution finally.

2.which feature would you like to highlight with this example?
The feature of this example is to demonstrate the fabulous parallel performance of taichi and prove that taichi can not only be applied in CFD, but also is suitable for various fields required parallel computing such as evolutionary algorithm (EA).

3.where to find the reference if the user wants to learn more about this algorithm?
Here are sone paper for differential algorithm (DE) and evolutionary algorithm (EA):
[1]. Differential Evolution – A Simple and Efficient Heuristic for global Optimization over Continuous Spaces
[2]. Application of differential evolution algorithm in static and dynamic economic or emission dispatch problem: A review

**Comment by neozhaoliang** [comment]
@Nanase-Nishino Hi. Would you please include that information as comments in your example so that users can see it? Thanks!

File: python/taichi/examples/machine_learning/differential_evolution.py [hunk 1]
```diff
@@ -1,3 +1,18 @@
+"""
+information about this example
+1.what does this script animate?
+This script reflects the optimization process of differential evolution. The black points are the search agents and they find out the minimum solution finally.
```
Comment (neozhaoliang) [inline python/taichi/examples/machine_learning/differential_evolution.py:4]:
```suggestion
"""
This script shows the optimization process of differential evolution. The black points are the search agents, and they finally find the minimum solution


.
```

File: python/taichi/examples/machine_learning/differential_evolution.py
Comment (Nanase-Nishino) [inline python/taichi/examples/machine_learning/differential_evolution.py:1]: This script reflects the optimization process of differential evolution. The black points are the search agents and they find out the minimum solution finally. The feature of this example is to demonstrate the fabulous parallel performance of taichi and prove that taichi can not only be applied in CFD, but also is suitable for various fields required parallel computing such as evolutionary algorithm (EA).
If anyone has more interest in DE and EA,  please contact me or review these papers:
[1]. Differential Evolution – A Simple and Efficient Heuristic for global Optimization over Continuous Spaces
[2]. Application of differential evolution algorithm in static and dynamic economic or emission dispatch problem: A review

File: python/taichi/examples/machine_learning/differential_evolution.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nanase-Nishino) [reply python/taichi/examples/machine_learning/differential_evolution.py]: > Hmm, I think a short description is just fine, here is an example contributed by the community:
> 
> https://github.com/taichi-dev/taichi/blob/master/python/taichi/examples/simulation/two_stream_instability.py

ok!!! Thanks for ur sincere and patient guidence!

File: python/taichi/examples/machine_learning/differential_evolution.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+            _trial[j] = _lb[j]
+
+
+@ti.func
+def f1(_fit: ti.template(), _pop: ti.template()):
+    _search_num, _dim = _pop.shape
+    for ii in range(_search_num):
+        cur = 0.0
+        for j in range(_dim):
+            cur += ti.pow(a=_pop[ii, j], b=2)
```
[the remark this answers is not in the document]
Comment (Nanase-Nishino) [reply python/taichi/examples/machine_learning/differential_evolution.py:42]: thanks for ur review! I can run this code in my local environmen without this TypeErrort: version 1.2.1, llvm 10.0.0, commit 12ab828a, win, python 3.8.10.

[the remark this answers is not in the document]
Comment (neozhaoliang) [reply python/taichi/examples/machine_learning/differential_evolution.py:42]: Would you please test it using a newer version, like v1.5.0?

[the remark this answers is not in the document]
Comment (neozhaoliang) [reply python/taichi/examples/machine_learning/differential_evolution.py:42]: I see; it was my bad; the code runs fine. But still, it's not a common way to use keyword args in the `pow` function`, something like `pow(x, 3.0)` would be more natural.

[the remark this answers is not in the document]
Comment (Nanase-Nishino) [reply python/taichi/examples/machine_learning/differential_evolution.py:42]: ok!! thanks for ur guidence! Please forgive me who is a fresh coder.

**Review by neozhaoliang — APPROVED** [review]
Thanks a lot!

**Comment by Nanase-Nishino** [comment]
> Thanks a lot!

thanks for ur guidence!! As a fresh coder, I learn a lot!!!

**Pull request merged**